com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ColladaUtil.java

```diff
@@ -571,7 +571,56 @@ private static XMLNode getFirstNodeWithGeoemtry(Collection<XMLVisualScene> scene
         return null;
     }
 
-    public static void loadMesh(XMLCOLLADA collada, Rig.MeshSet.Builder meshSetBuilder, boolean optimize) throws IOException, XMLStreamException, LoaderException {
+    private static float[] toFloatArray(List<Float> a) {
+        float[] floatArray = new float[a.size()];
+        for (int i = 0 ; i < a.size(); i++) {
+            floatArray[i] = (float) a.get(i);
+        }
+        return floatArray;
+    }
+
+    private static int[] toIntArray(List<Integer> a) {
+        return a.stream().mapToInt(x -> x).toArray();
+    }
+
+    private static ModelImporter.Mesh createModelImporterMesh(List<Float> position_list,
```

Comment:
It was easier to convert a collada mesh into a ModelImporter.Mesh, and let that class do the splitting.